cmd/run/run_test.go

```diff
@@ -563,10 +564,20 @@ func TestHTTPServerWithCORS(t *testing.T) {
 	}
 }
 
+// helper function to obtain default config and extra tcp port
+func obtainDefaultConfigWithExtraPort(t *testing.T) (*Config, int) {
```

Comment:
Mind just adding a testutil to the `pkg/testutils` package that can be used to obtain a random port and then using it directly instead of a very specific helper method for this? For example,

```go
// package pkg/testutils/testutils.go

func RandomTCPPort() (int, error) {
    	l, err := net.Listen("tcp", "")
         if err != nil {
             return -1, err
         }
	defer l.Close()
	
        return l.Addr().(*net.TCPAddr).Port, nil
}
```

```go
// package cmd/run/run_test.go

func TestBuildServerWithOIDCAuthentication(t *testing.T) {
    cfg := MustDefaultConfigWithRandomPorts()

    oidcPort, err := testutils.RandomTCPPort()
    require.NoError(t, err)

    localOIDCServerURL := fmt.Sprintf("http://localhost:%d", oidcPort)
```

Reply:
The example above will not work because we have released the port when we exit MustDefaultConfigWithRandomPorts() and testutils.RandomTCPPort(). Thus, it is entirely possible that we use the same port for oidcPort and one of either HTTP/GRPC port.

Reply:
`defer l.Close()` releases the port. The code you have currently is functionally the same as the code I proposed above.

Your concern is valid, but it's also already present in https://github.com/openfga/openfga/blob/main/cmd/run/run.go#L305-L320

It's possible we re-use a returned port in other code, but in practice we haven't experienced it (AFAIK).

Reply:
The code that is proposed will make it worse.  Previously, https://github.com/openfga/openfga/blob/main/cmd/run/run.go#L305-L320 will obtain two random ports before releasing.  Therefore, these two ports are guarantee not to collide and you will need to use have another caller trying to obtain the ports immediately for collision.  However, if we separate out via RandomTCPPort, you will release either the extra random port OR release both ports for default config immediately before obtaining a new port.  Thus, the chance of collision is much higher.

Reply:
Discussed offline, we can have RandomTCPPort but we need to return a function that can be called to release the port.  Otherwise, the ports may be re-used